src/main/java/Duke.java

```diff
@@ -1,10 +1,205 @@
+                            DIVIDER);
+                } else {
+                    addDeadline(paramAndBy[0], paramAndBy[1]);
+                }
+            }
+            break;
+        case COMMAND_EVENT_WORD:
+            if (canStore()) {
+                final String[] paramAndFromTo = processEvent(param);
+                if (paramAndFromTo == null) {
+                    System.out.println(DIVIDER +
+                            "Please enter event as \"event [task] /from [date] /to [date]\". \n" +
+                            DIVIDER);
+                } else {
+                    addEvent(paramAndFromTo[0], paramAndFromTo[1], paramAndFromTo[2]);
+                }
+            }
+            break;
+        }
+    }
+
+    private static boolean canStore() {
+        if (counter < CAPACITY) {
+            return true;
+        }
+        printErrorForStorage();
+        return false;
+    }
+
+
+    private static void initTasks() {
+        allTasks = new Task[CAPACITY];
+        counter = 0;
+    }
+
+    private static String[] processCommand(String userCommand) {
+        final String[] split = userCommand.trim().split("\\s+", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""};
+    }
+
+    private static String[] processDeadline(String param) {
+        String[] split = param.trim().split("\\s/by\\s", 2);
+        return split.length == 2 ? split : null;
+    }
+
+    private static String[] processEvent(String param) {
+        String[] split = param.trim().split("\\s/from\\s|\\s/to\\s", 3);
+        return split.length == 3 ? split : null;
+    }
+
+    private static void markNotDone(String param, Task[] store, int counter) {
+        try {
+            int idx = Integer.parseInt(param) - 1;
+            if (idx < 0 || idx >= counter) {
+                printErrorForIdx(counter);
+                return;
+            }
+            store[idx].setDone(false);
+            System.out.println(DIVIDER +
+                    "OK, I've marked this task as not done yet:\n" +
+                    "[ ] " + store[idx].getDescription() + "\n" +
+                    DIVIDER);
+        } catch (NumberFormatException ex) {
+            printErrorForIdx(counter);
+        }
+    }
+
+    private static void markDone(String param, Task[] store, int counter) {
+        try {
+            int idx = Integer.parseInt(param) - 1;
+            if (idx < 0 || idx >= counter) {
+                printErrorForIdx(counter);
+                return;
+            }
+            store[idx].setDone(true);
+            System.out.println(DIVIDER + "Nice!, I've marked this task as done:\n" +
+                    store[idx] + "\n" + DIVIDER);
+        } catch (NumberFormatException ex) {
+            printErrorForIdx(counter);
+        }
+    }
+
+    private static void addToDo(String param) {
+        allTasks[counter] = new ToDo(param);
+        printAddMessage(allTasks[counter]);
+        counter++;
```

Comment:
Maybe could do a static counter variable in the Task Class instead of Duke Class. Or is there any reason you chose to do it this way?